HDSAviz/data_processing.py

```diff
@@ -0,0 +1,218 @@
+    and returns a pandas dataframe with its contents.
+    The first line of the input should contain headers
+    corresponding to the column names.
+
+    Parameters:
+    ----------
+    filename  : String with the complete filename, including
+                absolute or relative path.
+    numrows   : Integer number of rows of the file to read.
+                If you don't specify this parameter all rows
+                will be read.
+    drop      : A list of strings indicating which (if any)
+                of the named columns you do not want to include
+                in the resulting dataframe. (ex. ['cats', 'dogs'],
+                default is not to drop any rows)
+    sep       : String indicating the column separator in the
+                file (optional, default = ',').
+
+    Returns:
+    --------
+    df : A pandas dataframe with the contents of the file,
+         limited to the number of rows specified and without the
+         columns named in "drop".
+    """
+    df = pd.read_csv(filename, sep=sep, nrows=numrows)
+    if not drop:
+        df.drop(drop, axis=1, inplace=True)
+
+    return df
+
+
+def get_params(numrows=None, drop=['End_time', 'Oxygen']):
+    """
+    NOTE: This function is specific to our lignin modeling dataset
+
+    Returns a pandas dataframe with all the parameters analyzed in
+    the sensitivity analysis, but not additional parameters like
+    end time and oxygen content.  If you would like all of the
+    parameters (even those not analyzed for sensitivity) then pass
+    drop=False.
+
+    Parameters:
+    ----------
+    numrows : the number of rows of the input_parameters file to read
+              (default is to read all rows).
+    drop    : a list of strings for which parameters you do not want to
+              include in the returned dataframe.  If you want all params
+              then pass drop=False.
+
+    Returns:
+    -------
+    A pandas dataframe
+
+    """
+    return read_file('../../HDSAviz_data/input_parameters.csv',
+                     numrows=numrows, drop=drop)
+
+
+def get_results(numrows=None, drop=['light_aromatic_C-C',
+                                    'light_aromatic_methoxyl']):
+    """
+    NOTE: This function is specific to our lignin modeling dataset
+
+    Returns a pandas dataframe with the results of running all of the
+    simulations for the parameters sets in `input_parameters.csv`. This
+    function drops two unused functional groups from the results file.
+
+    Parameters:
+    ----------
+    numrows : the number of rows of the input_parameters file to read
+              (default is to read all rows).
+    drop    : a list of strings for which output measures to drop from
+              the returned dataframe.  If you want all outputs use
+              drop=False.
+    """
+    return read_file('../../HDSAviz_data/results.csv',
+                     numrows=numrows, drop=drop)
+
+
+def get_sa(path='../../HDSAviz_data/'):
```

Comment:
# Naming convention: 
get_sa name seems to be a bit harder name to grasp on what it might  be doing. 
it seems to mean like get sensitivity analysis, and one could assume its going to give a final plot or a conclusive result but its returning out some dictionary of data frames. 
It would be great if its output is given a name like dict_of_SA_dataframes = sens[key] or something. 
also it would be nice to have easy to grasp names for sens[key][0]  like dict_of_SA_dfs_wo_sec_ord_int that could be reused.